[BE-3150]: Added firebase deployment doc (#424)
appcircleio/appcircle-docusaurus: docs/workflows/common-workflow-steps/firebase-deloyment.md

| [**Flutter Install**](/workflows/flutter-specific-workflow-steps/flutter-install) | This step installs the [Flutter SDK](https://docs.flutter.dev/deployment/web#building-the-app-for-release). If no version is specified, it installs the latest stable version. |

### Input Variables

You can find all the parameters required for this step in the table below, with their descriptions in detail.

:::warning

Do not hard-code sensitive variables, such as tokens and API keys, directly to the parameters in the step.

We recommend using [**Environment Variables**](/environment-variables/managing-variables) groups for such sensitive variables.

:::

| Variable Name                            | Description                                                         | Status           |
|------------------------------------------|---------------------------------------------------------------------|------------------|
| `$AC_FIREBASE_VERSION`                   | Firebase version to be used. Enter v11.11.0 for a specific version. | Required |
| `$AC_FIREBASE_PROJECT_PATH`              | The directory containing your `firebase.json` file.                 | Required |

> | `$AC_FIREBASE_TOKEN`                     | A refresh token that's printed when you authenticate with `firebase login:ci` command. **Either select Firebase token or Google Service account**. | Optional |

Reviewer: ```suggestion
| `$AC_FIREBASE_TOKEN`                     | A refresh token is generated when you authenticate using the `firebase login:ci` command. Choose either a **Either select Firebase token or Google Service account**.. | Optional |
```

> | `$GOOGLE_APPLICATION_CREDENTIALS`        | Path of Google Service Account JSON. Upload service account as a file to your environment group and name it `GOOGLE_APPLICATION_CREDENTIALS`. **Either select Firebase token or Google Service account**. | Optional |

Reviewer: ```suggestion
| `$GOOGLE_APPLICATION_CREDENTIALS`        | Specify the path to your Google Service Account JSON file. Upload this file to your environment group under the name `GOOGLE_APPLICATION_CREDENTIALS`. Choose either a **Firebase token**  or a **Google Service account**. | Optional |
```
